chef/cookbooks/barclamp/libraries/nic.rb

```diff
@@ -566,7 +566,6 @@ def add_slave(slave)
       usurp(slave)
```

Comment:
> usurp() removes addresses, routes and flushes the device

Right, but more importantly it also assigns those addresses and routes to the parent interface (the bond).  Which I think is what @vuntz worried about. Because "wicked ifdown" is removing the interface, there is nothing that "ursurp" can assing to the bond. However is might not really be an issue.

@vuntz: The bond will get its addresses and routes assigned further down in the recipe anyway (https://github.com/s-t-e-v-e-n-k/crowbar-core/blob/006506f42bf009f324ea25aa4538bbbcfca0a91b/chef/cookbooks/network/recipes/default.rb#L436). In the case where the bond is newly created that would also be the time when the interface is put "up" for the first time. If the slave interface had any kind of cluster VIP assigned (which would mean there was  a cluster setup without bonding. Somewhat unlikely) that would also be the place where the VIP would be taken away from the bond.
I might be missing something but I can't currently think of a case where the usurp would add any IP to the bond that the later code would not add. Or am I missing your point?

>  which might be required if some interface was configured manually and there is no wicked configuration for it -- however, I can test this with the usurp() call removed, and see what happens.

I don't think this is needed currently.